src/app/clusters/on-off-server/on-off-server.cpp

```diff
@@ -262,8 +265,228 @@ void emberAfOnOffClusterServerInitCallback(EndpointId endpoint)
     emberAfPluginOnOffClusterServerPostInitCallback(endpoint);
 }
 
+bool OnOffServer::offCommand()
+{
+    EmberAfStatus status = setOnOffValue(emberAfCurrentEndpoint(), Commands::Off::Id, false);
+#ifdef EMBER_AF_PLUGIN_ZLL_ON_OFF_SERVER
+    if (status == EMBER_ZCL_STATUS_SUCCESS)
+    {
+        emberAfPluginZllOnOffServerOffZllExtensions(emberAfCurrentCommand());
+    }
+#endif
+    emberAfSendImmediateDefaultResponse(status);
+    return true;
+}
+
+bool OnOffServer::onCommand()
+{
+    EmberAfStatus status = setOnOffValue(emberAfCurrentEndpoint(), Commands::On::Id, false);
+
+#ifdef EMBER_AF_PLUGIN_ZLL_ON_OFF_SERVER
+    if (status == EMBER_ZCL_STATUS_SUCCESS)
+    {
+        emberAfPluginZllOnOffServerOnZllExtensions(emberAfCurrentCommand());
+    }
+#endif
+
+    emberAfSendImmediateDefaultResponse(status);
+    return true;
+}
+
+bool OnOffServer::toggleCommand()
+{
+    EmberAfStatus status = setOnOffValue(emberAfCurrentEndpoint(), Commands::Toggle::Id, false);
```

Comment:
this looks like 'setting to off' instead of toggling. Is this correct?

Reply:
setOnOffValue had the logic to process On, Off and Toggle so it executes it properly.